bool.go

```diff
@@ -35,8 +36,8 @@ func NewArrayB(data []bool, shape ...int) (a *Arrayb) {
 			}
 			return
 		}
-		sz *= uint64(v)
-		sh[i] = uint64(v)
+		sz *= v
+		sh[i] = v
```

Comment:
Casting isn't needed, so we can pull this assignment outside the for loop: `copy(sh, shape)`